### PR TITLE
linting: add `gofumpt`

### PR DESCRIPTION
**Pull request by geoknee**
`gofumpt` is a more opinionated linter which (among other things) will complain if there are blank lines at the start or end of a function.

We have a few cases of that, and this rule often comes up during PR review. With this change, the rule is enforced in CI and applied on save for VSCode users. 

Questions:
* We may with to harmonize this with the approach in other optimism repos.

### CODE REVIEW

**Review by mds1 — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (trianglesphere) [inline .vscode/settings.json:1]: TBH I don't love pulling in editor specific config to our repos.

File: .golangci.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ run:
 linters:
   # also see https://golangci-lint.run/usage/linters/#enabled-by-default
   enable:
-    - goimports
+    - gofumpt
```
Comment (trianglesphere) [inline .golangci.yml:7]: Is `gofumpt` a superset of `goimports`? I know it's a superset of `gofmt`, but `goimports` is also a superset of `gofmt`

**Review by trianglesphere — COMMENTED** [review]
I like integrating this via golangci lint. IDK if I'd want to use it to replace gofmt/goimports in people's normal formatting

File: .golangci.yml (hunk 2 above)
Comment (geoknee) [reply .golangci.yml:7]: Good question. I think the answer is actually "no (but apparently don't worry about it?)" https://github.com/mvdan/gofumpt?tab=readme-ov-file#frequently-asked-questions

Reading that link, looks like we can enable both if we want to.

File: .vscode/settings.json (hunk 1 above)
Comment (geoknee) [reply .vscode/settings.json:1]: I get that feeling a tiny bit, but wonder if the benefits outweigh the drawbacks. My overall aim is we talk about formatting changes as little as possible. This config makes it less likely for a developer to be surprised by linting failures on CI. 

I don't feel super strongly though. We can rip this out and let each dev sort their editor out by themselves.

Comment (mds1) [reply .vscode/settings.json:1]: > TBH I don't love pulling in editor specific config to our repos.

I thought this too but noticed we also have a `.vscode` in the monorepo from a few years back: https://github.com/ethereum-optimism/optimism/pull/1109

If we do remove this, we can add `.vscode` to the gitignore then explain how to set this up in CONTRIBUTING. Can be VSCode only instructions for now that are expanded over time

File: .golangci.yml (hunk 2 above)
Comment (trianglesphere) [reply .golangci.yml:7]: Yea, let's enable both

File: .vscode/settings.json (hunk 1 above)
Comment (geoknee) [reply .vscode/settings.json:1]: Sounds like a good compromise to me.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -168,4 +168,18 @@ go run ./op-chain-ops/cmd/registry-data \
...
   --bytecodes-dir=$SUPERCHAIN_REPO/superchain/extra/bytecodes \
   --output=$SUPERCHAIN_REPO/superchain/extra/genesis/$SUPERCHAIN_TARGET/$CHAIN_NAME.json.gz
-```
\ No newline at end of file
+```
+
+## Setting up your editor for formatting and linting
+If you use VSCode, you can place the following in a `settings.json` file in the gitignored `.vscode` directory:
+
+```
```
Comment (mds1) [inline CONTRIBUTING.md:176]: nit

Comment (mds1) [inline CONTRIBUTING.md:176]: nit
```suggestion
```json
```

File: .vscode/settings.json (hunk 1 above)
Comment (mds1) [reply .vscode/settings.json:1]: LGTM, let's also delete this existing `.vscode/settings.json` that's added in this PR

Comment (geoknee) [reply .vscode/settings.json:1]: [b97eb4e](https://github.com/ethereum-optimism/superchain-registry/pull/105/commits/b97eb4e5e050f8028fa6414e59b24288ae089758)

Comment (geoknee) [reply .vscode/settings.json]: [a3e513b](https://github.com/ethereum-optimism/superchain-registry/pull/105/commits/a3e513b23c1823faed9b3fcee450fb9833167155)

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**